File: mynlp-core/src/main/java/com/mayabot/nlp/utils/DownloadUtils.java
package com.mayabot.nlp.utils;

import com.google.common.io.ByteStreams;
import com.google.common.io.Files;
import com.google.common.io.Resources;

import java.io.*;
import java.net.URL;
import java.util.zip.ZipEntry;
import java.util.zip.ZipInputStream;

public class DownloadUtils {

    public static void main(String[] args) throws Exception {
        download("http://cdn.mayabot.com/nlp/hotel-test.txt.zip", new File("data/test.text.zip"));
        unzip(new File("data/test.text.zip"));
    }

    /**
     * 下载文件
     *
     * @param url
     * @param file
     */
    public static void download(String url, File file) throws IOException {
        BufferedOutputStream outputStream = new BufferedOutputStream(new FileOutputStream(file));
        Resources.copy(new URL(url), outputStream);

        outputStream.flush();
        outputStream.close();
    }

    /**
     * unzip file
     *
     * @param file
     * @throws Exception
     */
    public static void unzip(File file) throws Exception {

        try (ZipInputStream zipInputStream = new ZipInputStream(Files.asByteSource(file).openBufferedStream())) {
            ZipEntry entry = null;
            while ((entry = zipInputStream.getNextEntry()) != null) {
                String name = entry.getName();
                File toFile = new File(file.getParent(), name);
                OutputStream outputStream = Files.asByteSink(toFile).openBufferedStream();
                ByteStreams.copy(zipInputStream, outputStream);

                outputStream.flush();
                outputStream.close();
            }

        }

    }
}

File: mynlp/src/main/java/com/mayabot/nlp/utils/DownloadUtils.kt
package com.mayabot.nlp.utils

import java.io.File
import java.io.IOException
import java.net.URL
import java.util.zip.ZipInputStream

object DownloadUtils {

    /**
     * 下载文件
     *
     * @param url
     * @param file
     */
    @Throws(IOException::class)
    @JvmStatic
    fun download(url: String, file: File) {
        //先完全读入到内存中去。然后一次性写入文件
        file.writeBytes(URL(url).readBytes())
    }

    /**
     * unzip file
     *
     * @param file
     * @throws Exception
     */
    @Throws(Exception::class)
    @JvmStatic
    fun unzip(file: File) {

        ZipInputStream(file.inputStream().buffered()).use { zipInputStream ->
            var entry = zipInputStream.nextEntry

            while (entry != null) {
                val name = entry.name

                File(file.parent, name).outputStream().buffered().use {
                    zipInputStream.copyTo(it)
                }

                entry = zipInputStream.nextEntry
            }

        }

    }
}
